src/components/course/course-header/CourseImportantDates.jsx

```diff
@@ -0,0 +1,112 @@
+  getSoonestEarliestPossibleExpirationData,
+  useCourseMetadata,
+  useRedeemablePolicies,
+} from '../../app/data';
+
+const messages = defineMessages({
+  importantDates: {
+    id: 'enterprise.course.about.page.important-dates.title',
+    defaultMessage: 'Important dates',
+    description: 'Title for the important dates section on the course about page',
+  },
+  enrollByDate: {
+    id: 'enterprise.course.about.page.important-dates.enroll-by-date',
+    defaultMessage: 'Enroll-by date',
+    description: 'Enroll-by date for the important dates section on the course about page',
+  },
+  courseStarts: {
+    id: 'enterprise.course.about.page.important-dates.course-starts',
+    defaultMessage: 'Course starts',
+    description: 'Course starts for the important dates section on the course about page in future tense',
+  },
+  courseStarted: {
+    id: 'enterprise.course.about.page.important-dates.course-started',
+    defaultMessage: 'Course started',
+    description: 'Course started the important dates section on the course about page in past tense',
+  },
+});
+
+const CourseImportantDates = () => {
+  const { courseKey } = useParams();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: courseMetadata } = useCourseMetadata();
+  const intl = useIntl();
+  const {
+    allocatedCourseRunAssignments,
+    allocatedCourseRunAssignmentKeys,
+    hasAssignedCourseRuns,
+  } = determineAllocatedCourseRunAssignmentsForCourse({
+    redeemableLearnerCreditPolicies,
+    courseKey,
+  });
+
+  const [searchParams] = useSearchParams();
+  const courseRunKey = searchParams.get('course_run_key')?.replaceAll(' ', '+');
+
+  if (!hasAssignedCourseRuns || (!!courseRunKey && !allocatedCourseRunAssignmentKeys.includes(courseRunKey))) {
+    return null;
+  }
+
+  const { date, soonestExpirationDateData } = getSoonestEarliestPossibleExpirationData({
+    assignmentObjectArray: allocatedCourseRunAssignments,
+    dateFormat: `${DATE_FORMAT} h:mm A`,
+  });
+  const soonestExpiringAllocatedAssignmentCourseStartDate = courseMetadata.availableCourseRuns.find(
+    (courseRun) => courseRun.key === soonestExpirationDateData?.contentKey,
+  )?.start;
+
+  const enrollByDate = date;
+  const courseStartDate = dayjs(soonestExpiringAllocatedAssignmentCourseStartDate).format(DATE_FORMAT);
+  const hasCourseStarted = dayjs(courseStartDate).isBefore(dayjs());
+
+  return (
+    <Stack gap={2} className="mt-4">
+      <h3>{intl.formatMessage(messages.importantDates)}</h3>
+      <section>
+        {enrollByDate && (
+          <Row>
+            <Col>
+              <Stack direction="horizontal" gap={2}>
+                <Icon src={Calendar} />
+                {intl.formatMessage(messages.enrollByDate)}
+              </Stack>
+            </Col>
+            <Col>
+              {enrollByDate}
+            </Col>
+          </Row>
+        )}
+        {enrollByDate && courseStartDate && <hr />}
```

Comment:
[suggestion] I think it might make sense to rely on border styling versus the `hr`. Rationale: styling the border such that each date has a `.border-bottom` would ensure the styling is propagated for any other future dates/items that might be included here, too (otherwise, the `hr` would need to replicate in numerous places).

For example, the `Row` could depend on `.border-bottom`, but then you'd need to pass `<Row className="course-important-date border-bottom px-0 py-2">` and `Col className="px-0"` for the border to look as expected, width wise.

Then, to remove the last date's border:

```scss
// _CourseHeader.scss
.course-header {
  // ...
  .assignments-important-dates {
    .course-important-date:last-child {
      border-bottom: none !important; // !important ensures override of the default border-bottom classname for the last child
    }
  }
}
```